main/reference/zoe-api/zoe.md

```diff
@@ -286,73 +286,63 @@ const { creatorFacet, publicFacet, creatorInvitation } = await E(Zoe).startInsta
 
-The invocation normally includes a **proposal** (the
-rules under which they want to exercise the offer) and **payments** that correspond 
-to the **give** property of the **proposal**. The payments will be escrowed by Zoe. If
-either the **proposal** or **payments** are empty, indicate this by
-omitting that argument or passing **undefined**, instead of passing an
-empty record.
-
-The optional **exit**'s value should be an **exitRule**, an object with three possible keys for
-key:value pairs:
-
-- **onDemand: null**: (Default) The offering party can cancel on demand.
-- **waived: null**: The offering party can't cancel and relies entirely on the smart contract to promptly finish their offer.
-- **afterDeadline**: The offer is automatically cancelled after a deadline, as determined 
-  by its **timer** and **deadline** properties.
-  **timer** must be a timer, and **deadline** must be a [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) value interpreted with respect to the timer.
-  Some example timers use Unix epoch time, while others count block height.
+**proposal** must be either `undefined` or a record with **give**, **want**, and/or **exit** properties
+respectively expressing conditions regarding what assets will be given, what must be
+received in exchange, and an exit rule defining how/when the offer can be canceled.
 
 ```js
 const myProposal = harden({
-  give: { Asset: quatloos(4n)},
-  want: { Price: moola(15n) },
-  exit: { afterDeadline: {
-    timer,
-    deadline: 100n
-  }}
-})
+  give: { Asset: AmountMath.make(quatloosBrand, 4n) },
+  want: { Price: AmountMath.make(moolaBrand, 15n) },
+  exit: { onDemand: null },
+});
 ```
 
-**paymentKeywordRecord** is a record with **[Keywords](./zoe-data-types.md#keyword)** as keys and with
- values of the actual **payments** to be escrowed. A **payment** is
- expected for every entry under **give**.
+**give** and **want** use **[Keywords](./zoe-data-types.md#keyword)** defined by the contract.
+In the example above, "Asset" and "Price" are the Keywords. However, in an auction contract,
+the Keywords might be "Asset" and "Bid".
 
- **offer()** returns a promise for a **UserSeat**.
- 
-```js
-const paymentKeywordRecord = {
-  Asset : quatloosPayment,
-  Price : moolaPayment
-};
-```
+**exit** specifies how the offer can be can cancelled. It must conform to one of three shapes:
+- `{ onDemand: null }`: (Default) The offering party can cancel on demand.
+- `{ waived: null }`: The offering party can't cancel and relies entirely on the smart contract to promptly finish their offer.
+- `{ afterDeadline: deadlineDetails }`: The offer is automatically cancelled after a deadline,
+  as determined by its **timer** and **deadline** properties.
+  **timer** must be a timer, and **deadline** must be a [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) value interpreted with respect to the timer.
```

Comment:
deadline must be a bigint? I have recent experience using a `{ timerBrand, absValue }` record.

Reply:
[deadline must be a bigInt](https://docs.agoric.com/reference/zoe-api/zoe.html#e-zoe-offer-invitation-proposal-paymentkeywordrecord-offerargs)

[TimerService also uses bigInts](https://docs.agoric.com/reference/repl/timerServices.html#timerservice-objects)

Reply:
> deadline must be a bigInt

Well, those docs are wrong (which is somewhat inevitable until we sync them by machine https://github.com/Agoric/agoric-sdk/issues/4291). As I say, I have recent experience to the contrary. Plus, the code says otherwise:

```js
      afterDeadline: {
        timer: M.eref(TimerShape),
        deadline: TimestampShape,
      },
```
https://github.com/Agoric/agoric-sdk/blob/6d9ae67c57499d7139dfce019b8c76f9ecdb825f/packages/zoe/src/typeGuards.js#L82-L85

```js
export const TimestampShape = M.or(TimestampRecordShape, TimestampValueShape);
```
https://github.com/Agoric/agoric-sdk/blob/6d9ae67c57499d7139dfce019b8c76f9ecdb825f/packages/time/src/typeGuards.js#L17C1-L17C79

Reply:
Fixed.